### PR TITLE
Support main media element in AMP

### PR DESCRIPTION
**Pull request by SiAdcock**
## What does this change?

Adds support for main media element in AMP

## Why?

A picture says a thousand words

## Screenshot

screen shot 2018-12-24 at 15 06 06

### CODE REVIEW

File: packages/frontend/lib/model/extract-capi.ts [hunk 1]
```diff
@@ -232,6 +232,11 @@ export const extract = (data: {}): CAPIType => {
                 : undefined,
             email: 'none',
         },
+        mainMediaElements: getArray<ImageBlockElement>(
```
Comment (nicl) [inline packages/frontend/lib/model/extract-capi.ts]: Can the main element be a non-image? I.e. video etc.?

File: packages/frontend/lib/content.d.ts [hunk 2]
```diff
@@ -14,8 +14,10 @@ interface RichLinkBlockElement {
 interface ImageBlockElement {
     _type: 'model.dotcomrendering.pageElements.ImageBlockElement';
     media: { allImages: Image[] };
-    data: { alt: string; credit: string; caption?: string };
+    data: { alt: string; credit: string; caption?: string; copyright?: string; };
```
Comment (nicl) [inline packages/frontend/lib/content.d.ts]: Just checking, but is this needed for non-main images too?

Comment (SiAdcock) [reply packages/frontend/lib/content.d.ts]: I have seen it on non-main images ([example](https://www.theguardian.com/games/2019/jan/04/rainbow-arcade-lgbtq-video-games-exhibition-world-of-warcraft-nintendo-berlin.json?guui))

File: packages/frontend/lib/model/extract-capi.ts (hunk 1 above)
Comment (SiAdcock) [reply packages/frontend/lib/model/extract-capi.ts]: Good point. I'll change this to an array of `CAPIElement`s for now, until we determine exactly which types they could take.

**Comment by SiAdcock** [comment]
Failing gracefully if the main media is not an image means nothing renders. I think this is fine; at least 
this change represents an improvement.

**Pull request merged**